File: matlab-api/+smalltargetmotiondetectors/+api/get_visualize_handle.m
function hVisualization = get_visualize_handle(className, showThreshold)
    % GET_VISUALIZE_HANDLE returns a handle to a visualization object based
    % on the given class name.
    % 
    % Syntax:
    %   hVisualization = get_visualize_handle(className)
    % 
    % Input:
    %   className - Name of the visualization class. 
    % If empty, a default visualization object is created.
    % 
    % Output:
    %   hVisualization - Handle to the visualization object.
    % 
    % Example:
    %   % Creates a default visualization object
    %   hVis = get_visualize_handle(); 
    %   % Creates a visualization object of class 'MyVisualizationClass'
    %   hVis = get_visualize_handle('MyVisualizationClass');

    % Get the full path of this file
    filePath = mfilename('fullpath');
    %   Find the index of '/matlab/+smalltargetmotiondetectors/'
    % in the file path
    indexPath = strfind(filePath, ...
        [filesep, 'matlab', filesep, '+smalltargetmotiondetectors', filesep]);
    % Add the path to the package containing the models
    addpath(filePath(1:indexPath(end)+7));

    import smalltargetmotiondetectors.util.iostream.*;

    if nargin == 2
        docIdx = strfind(className, '.');
        hVisualization = Visualization(...
            className(docIdx(end)+1:end), ...
            showThreshold);
    elseif nargin == 1
        docIdx = strfind(className, '.');
        hVisualization = Visualization(className(docIdx(end)+1:end));
    elseif nargin == 0
        hVisualization = Visualization();
    end

    hVisualization.create_fig_handle();
end

File: src/XT-TMP/api/get_visualize_handle.py
from ..util.iostream import Visualization

def get_visualize_handle(className=None, 
                         showThreshold=None, 
                         width = 8, 
                         height = 5, 
                         dpi = 100):
    """
    Returns a handle to a visualization object based on the given class name.

    Parameters:
        className (str): Name of the visualization class. If None, a default visualization object is created.
        showThreshold (bool): Whether to show threshold. Default is None.

    Returns:
        Visualization: Handle to the visualization object.
    """

    if className and showThreshold is not None:
        objVisualization = Visualization(className, showThreshold)
    elif className:
        objVisualization = Visualization(className)
    else:
        objVisualization = Visualization()
    
    objVisualization.create_fig_handle(width=width, height =height, dpi=dpi)

    return objVisualization

